server/src/main/java/org/elasticsearch/common/logging/ThrottlingAndHeaderWarningLogger.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.common.SuppressLoggerChecks;
+
+/**
+ * This class wraps both <code>HeaderWarningLogger</code> and <code>ThrottlingLogger</code>
+ * which is a common use case across Elasticsearch
+ */
+public class ThrottlingAndHeaderWarningLogger {
+    private final HeaderWarningLogger headerWarningLogger = new HeaderWarningLogger();
+    private final ThrottlingLogger throttlingLogger;
+
+    public ThrottlingAndHeaderWarningLogger(Logger loggerName) {
+        this.throttlingLogger = new ThrottlingLogger(LogManager.getLogger(loggerName));
+    }
+
+    /**
+     * Logs a message, adding a formatted warning message as a response header on the thread context.
+     */
+    //TODO fix this
+    @SuppressLoggerChecks(reason = "safely delegates to logger")
+    public void logAndAddWarning(String msg, Object... params) {
```

Comment:
true, it might be a bit odd. but the alternative is to another 'combination class' that would always log and emit a header warning.

Reply:
as discussed in person, i think this can be removed since we only want to expose the Deprecation logger and the Header warning logger. We want to avoid general usaage of this class and if a user wants to always header warn they can use that method and throttling to the main server log is something that that requires more consideration.